src/json.cpp

```diff
@@ -182,26 +182,26 @@ std::string JsonObject::str()
     }
 }
 
-void JsonObject::throw_error( std::string err, const std::string &name )
+void JsonObject::throw_error( const std::string &err, const std::string &name )
 {
     jsin->seek( verify_position( name, false ) );
     jsin->error( err );
 }
 
-void JsonArray::throw_error( std::string err )
+void JsonArray::throw_error( const std::string &err )
 {
     jsin->error( err );
 }
 
-void JsonArray::throw_error( std::string err, int idx )
+void JsonArray::throw_error( const std::string &err, const int &idx )
```

Comment:
Why would a reference to an int be better here?

Reply:
To not create a local copy if you don't need one.

Reply:
A local copy of an `int` is on the stack, while a reference is behind the scene an address that needs to be dereferenced. Given that an `int` usually has lower or equal bit width than an address, I would expect the local-copy to be faster to access?